File: code/client/economycraft/lib/classes/order.dart
import 'package:economycraft/classes/product.dart';
import 'package:economycraft/classes/company.dart';

class Order {
  final int id;
  final int productId;
  final int companyId;
  final int userId;
  final int quantity;
  final double payment;
  final String deliveryAddress;
  final DateTime orderTimeout;
  final DateTime createdAt;
  bool complete;
  bool received;

  Product? product;
  Company? company;

  Order({
    required this.id,
    required this.productId,
    required this.companyId,
    required this.userId,
    required this.quantity,
    required this.payment,
    required this.deliveryAddress,
    required this.orderTimeout,
    required this.createdAt,
    required this.complete,
    required this.received,
    this.product,
    this.company,
  });
}

File: code/server-processes/src/classes/classes/order.py
from datetime import datetime
from typing import Optional


class Order:
    
    def __init__(self, id: int, created_at: str, product_id: int, company_id: int, delivery_address: str, quantity: int, payment: float, order_timeout: str, complete: bool, user_id: int, received: bool,) -> None:
        self.id = id
        self.created_at = datetime.fromisoformat(created_at)
        self.product_id = product_id
        self.company_id = company_id
        self.delivery_address = delivery_address
        self.quantity = quantity
        self.payment = payment
        self.order_timeout = datetime.fromisoformat(order_timeout)
        self.complete = complete
        self.user_id = user_id
        self.received = received
        
        